src/lib/userAuthCall/1balance/index.ts

```diff
@@ -0,0 +1,145 @@
+import axios from "axios";
+import { BigNumber, providers } from "ethers";
+import { getAddress } from "ethers/lib/utils";
+
+import {
+  GELATO_RELAY_URL,
+  DEFAULT_DEADLINE_GAP,
+  getRelayAddress,
+  DEFAULT_INTERNAL_ERROR_MESSAGE,
+} from "../../../constants";
+import {
+  getEIP712Domain,
+  calculateDeadline,
+  getUserNonce,
+  signTypedDataV4,
+} from "../../../utils";
+import { PaymentType, RelayRequestOptions, RelayResponse } from "../../types";
+import { getFeeToken } from "../../../utils/getFeeToken";
+import { UserAuthSignature } from "../types";
+
+import {
+  EIP712_USER_AUTH_CALL_WITH_1BALANCE_TYPE_DATA,
+  UserAuthCallWith1BalancePayloadToSign,
+  UserAuthCallWith1BalanceRequest,
+  UserAuthCallWith1BalanceRequestOptionalParameters,
+  UserAuthCallWith1BalanceStruct,
+} from "./types";
+
+const getPayloadToSign = (
+  struct: UserAuthCallWith1BalanceStruct
+): UserAuthCallWith1BalancePayloadToSign => {
+  const verifyingContract = getRelayAddress(struct.chainId as number);
+  const domain = getEIP712Domain(
+    "GelatoRelay",
+    "1",
+    struct.chainId as number,
+    verifyingContract
+  );
```

Comment:
this should be centralized in an internal function to be reused by other 1balance methods